tests/TestCase/Controller/TrashControllerTest.php

```diff
@@ -0,0 +1,272 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2018 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace App\Test\TestCase\Controller;
+
+use App\Controller\TrashController;
+use BEdita\SDK\BEditaClient;
+use BEdita\SDK\BEditaClientException;
+use Cake\Http\Response;
+use Cake\Http\ServerRequest;
+use Cake\TestSuite\TestCase;
+
+class TrashControllerSample extends TrashController
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function render($view = null, $layout = null)
+    {
+        // do nothing
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function redirect($url, $status = 302) : Response
+    {
+        return new Response();
+    }
+
+    /**
+     * Set api client for controller
+     *
+     * @param BEditaClient $client the Api Client
+     * @return void
+     */
+    public function setApiClient(BEditaClient $client) : void
+    {
+        $this->apiClient = $client;
+    }
+}
+
+/**
+ * {@see \App\Controller\TrashController} Test Case
+ *
+ * @coversDefaultClass \App\Controller\TrashController
+ */
+class TrashControllerTest extends TestCase
+{
+    /**
+     * Test controller
+     *
+     * @var App\Test\TestCase\Controller\TrashControllerSample
+     */
+    public $Trash;
+
+    /**
+     * Test api client
+     *
+     * @var BEdita\SDK\BEditaClient
+     */
+    public $client;
+
+    /**
+     * Setup api client and auth
+     *
+     * @return void
+     */
+    private function setupApi() : void
+    {
+        $apiBaseUrl = getenv('BEDITA_API');
+        $apiKey = getenv('BEDITA_API_KEY');
+        $this->client = new BEditaClient($apiBaseUrl, $apiKey);
```

Comment:
This is already done in `AppController::initialize()` what about a `public getApiClient()` method to get client instead of a `setApiClient()`?
Actually..  `AppController::getApiClient()` could be used in components too....

Reply:
In this case we had to force to use another `apiClient` (local) in test context, that's why `setApiClient` could be useful... just for unit test